vsg/__main__.py

```diff
@@ -63,41 +62,40 @@ def get_predefined_styles():
     return lReturn
 
 
-def read_predefined_style(commandLineArguments):
+def read_predefined_style(sStyleName):
     '''
     Reads a predefined style file.
 
     Parameters :
 
-      commandLineArguments : (argparse object)
+      sStyleName : (string)
 
     Returns : (dictionary)
     '''
     dReturn = {}
-    if commandLineArguments.style:
-        sFileName = os.path.join(os.path.dirname(__file__), 'styles', commandLineArguments.style + '.yaml')
-        dReturn = open_configuration_file(sFileName, commandLineArguments)
+    sFileName = os.path.join(os.path.dirname(__file__), 'styles', sStyleName + '.yaml')
+    open_configuration_file(sFileName)
     return dReturn
     
 
-def open_configuration_file(sFileName, commandLineArguments):
+def open_configuration_file(sFileName, sJUnitFileName=None):
```

Comment:
Could you explain the name `sJUnitFileName`? What is its relation with `JUnit`?

Reply:
so `sJUnitFileName` is the file name passed to VSG via the `--junit` argument.

It is passed to the  `open_configuration_file` in case there is an error reading the configuration file.  If an error is detected and the `--junit` option was chosen, then the error is pushed to a JUnit XML file via the `write_invalid_configuration_junit_file` function.

Reply:
And what is JUnit per se? What is the idea behind writing such errors to JUnit XML?

Reply:
oh, JUnit?  As I understand it, JUnit is a Java unit test platform.  They came up with an XML format for showing errors in your Java testsuite.

Bamboo, the Continuous Integration (CI) tool we use at my work, had a parser for JUnit XML files.  Bamboo would parse these files and then indicate if there was a failure.

I wrote log parsers for all the tools we used:  Quartus, Spyglass, Questasim, Questa CDC etc... and generated these JUnit files.  It gave us finer control over whether we failed a job in Bamboo.

For example, I ran VSG on a simple entity file with two violations and I get this printed to the screen:
```mono
python bin/vsg -f debug/entity.vhd --junit blah.xml
================================================================================
File:  debug/entity.vhd
================================================================================
Phase 3 of 7... Reporting
Total Rules Checked: 198
Total Violations:    2
----------------------------+------------+--------------------------------------
  Rule                      |  line(s)   | Solution
----------------------------+------------+--------------------------------------
  whitespace_012            |          3 | Remove all but 1 blank lines below.
  entity_016                |          5 | Remove blank line(s) above "end entity" keywords.
----------------------------+------------+--------------------------------------
NOTE: Refer to online documentation at https://vhdl-style-guide.readthedocs.io/en/latest/index.html for more information.
```
If I add `--junit vsg.xml`, I also get this file:
```xml
<?xml version="1.0" ?>
<testsuite errors="0" hostname="Rimmer" failures="2" timestamp="2020-06-08T12:45:08" tests="1" time="0" name="vhdl-style-guide">
  <properties>
  </properties>
  <testcase name="debug/entity.vhd" time="0" classname="failure">
    <failure type="Failure">
      entity_016: 5 : Remove blank line(s) above &quot;end entity&quot; keywords.
      whitespace_012: 3 : Remove all but 1 blank lines below.
    </failure>
    <failure type="Failure">
      entity_016: 5 : Remove blank line(s) above &quot;end entity&quot; keywords.
      whitespace_012: 3 : Remove all but 1 blank lines below.
    </failure>
  </testcase>
  <system-out>
  </system-out>
  <system-err>
  </system-err>
</testsuite>

hmm...the data seems to be repeated.  I will have to fix that.

Anyway...the CI server will pull in the XML file and fail the run if it finds any failures.  You can then navigate to the errors and the data in the XML file will be displayed. 
```

Reply:
Ok, I understand. Is your log parsing tool open source?

Reply:
Those log parsers are not open source.  However, I did start a new side project  https://github.com/jeremiah-c-leary/eda-log-file-warning-suppressor that essentially performs the same function.  I currently only supports Mentor Graphics Precision, Microsemi Designer and Xilinx Vivado log files.

Maybe you could git a try and let me know what you think.